src/js/store/orgs/actions.ts

```diff
@@ -35,6 +35,10 @@ interface CommitEvent {
   type: 'SCRATCH_ORG_COMMIT_CHANGES' | 'SCRATCH_ORG_COMMIT_CHANGES_FAILED';
   payload: Org;
 }
+interface RefreshOrg {
+  type: 'REFRESH_ORG_STARTED' | 'REFRESH_ORG_SUCCEEDED' | 'REFRESH_ORG_FAILED';
+  payload: { org: Org; url: string; response?: any };
+}
```

Comment:
So, we have two sets of events here we're dealing with:

1. The request from the front-end to refresh the org (I'd suggest `SCRATCH_ORG_REFRESH_REQUESTED`, `SCRATCH_ORG_REFRESH_ACCEPTED`, and `SCRATCH_ORG_REFRESH_REJECTED`).
2. The events from the back-end telling us the result of the long-running action (which will be either `SCRATCH_ORG_REFRESH` or `SCRATCH_ORG_REFRESH_FAILED`).

For the former, the payload will not have a response. For the latter, it may.